org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java

```diff
@@ -1718,28 +1718,18 @@ protected Integer getIntSelectedRowKey(int tableId)
 
 	/**
      *  Get the keys of selected row/s based on layout defined in prepareTable
-     *  @deprecated
      *  @return IDs if selection present
      *  author ashley
      */
-    protected ArrayList<Object> getSelectedRowKeys()
-    {
-        ArrayList<Object> selectedDataList = new ArrayList<Object>();
-        Collection<Object> lsKeyValueOfSelectedRow = getSelectedRowInfo().keySet();
-        if (lsKeyValueOfSelectedRow.size() == 0)
-        {
-            return selectedDataList;
-        }
+	protected ArrayList<Object> getSelectedRowKeys() {
+		ArrayList<Object> selectedDataList = new ArrayList<Object>();
 
-        if (p_multipleSelection)
-        {        	
-        	for (Object key : lsKeyValueOfSelectedRow){        		
-        		selectedDataList.add(key);
-        	}
-        }
-
-        return selectedDataList;
-    }   //  getSelectedRowKeys
+		for (Map.Entry<Object, List<Object>> rowInfo : getSelectedRowInfo().entrySet()) {
+			if(rowInfo.getValue().get(0) instanceof IDColumn idColumn)
+				selectedDataList.add(idColumn.getRecord_ID());
+		}
+		return selectedDataList;
+	}  //  getSelectedRowKeys
```

Comment:
isn't the change needed here is just to remove the if (p_multipleSelection) condition ?

Reply:
Hi @hengsin, If we remove this condition if(p_multipleSelection) the feature where we can select multiple items and then create multiple records in the window will stop working, as we do in the Sales Order lines by selecting multiple products.

Reply:
No, I means it can be changed to as below instead:
'''
protected ArrayList<Object> getSelectedRowKeys()
{
        ArrayList<Object> selectedDataList = new ArrayList<Object>();
        Collection<Object> lsKeyValueOfSelectedRow = getSelectedRowInfo().keySet();
        for (Object key : lsKeyValueOfSelectedRow){        		
        	selectedDataList.add(key);
        }
       return selectedDataList;
}
'''

Reply:
Hi @hengsin, It would not change the problem, the situation is that when Single Selection we get the IDColumn(The keycolumn of Table set in MInfoWindow)
![2](https://github.com/idempiere/idempiere/assets/82419632/a92d7772-08c5-4d9e-a776-c7a0a77a91d4)
When getSelectedRowKey() we ends up here:
![1](https://github.com/idempiere/idempiere/assets/82419632/018bdaa3-c2fe-40a7-930a-119dfd95636d)
But if we have multiple selection and a keycolumn defined on MInfoColumn we are going to get its value,but i think we should always get the key of table defined on MInfoWindow.
The key below can be differente from IDColumn when a keyColumn on MInfoColumn is defined it will be returned
![3](https://github.com/idempiere/idempiere/assets/82419632/597c6702-c67b-44b3-86dc-4119e60f03e7)

Reply:
I understand where you are coming from here but that's not a backward compatible change.

Reply:
Hello @hengsin, do you think of any other way to deal with this behavior or do you have anything to suggest?

Reply:
hi @VitorHenri , it seems that method is only use by InfoWindow and is probably safe to ignore the backward compatibility issue. However, the changes above still need the following refinements:
- Follow return type of getSelectedRowKey, so it should return `List<T extends Serializable>` instead of `ArrayList<Object>`.
- Follow the implementation of getSelectedRowKey that cater for both IDColumn and UUIDColumn
- Remove the `@deprecated` javadoc tag.
- Remove author ashley from javadoc
- In javadoc, replace "Get the keys of selected row/s based on layout defined in prepareTable" with "Get the keys of selected rows" 
- In javadoc, replace "@return IDs if selection present" with "@return selected IDs or UUIDs"